wasmbinding/message_plugin.go

```diff
@@ -0,0 +1,282 @@
+package wasmbinding
+
```

Comment:
do we plan to keep this file?

Reply:
Is there a reason why we want a custom message dispatcher, rather than use the default message dispatcher with a custom CustomMsg encoder?

Reply:
I believe the default message dispatcher won't allow messages to the custom modules, right? or is that not the case?

Reply:
It allows custom message but the default behavior is to no-op. If you add something like this to wasmopts you can override the no-op behavior:
```
	wasmopts := []wasm.Option{wasmkeeper.WithMessageEncoders(&wasmkeeper.MessageEncoders{
		Custom: xwasm.DexCustomEncoder,
	})}
```
Of course we can change `DexCustomEncoder` to be something more generic